lib/puppet/provider/mount.rb

```diff
@@ -5,6 +5,9 @@
 module Puppet::Provider::Mount
   # This only works when the mount point is synced to the fstab.
   def mount
+    # Create the mountpoint directory if it does not exist and manage_mountpoint is true
+    Dir.mkdir(resource[:name]) if resource[:manage_mountpoint] && !File.exist?(resource[:name])
```

Comment:
The directory permissions here probably don't matter, but a default of `0500` is probably just fine. I don't think it's something we want to make configurable though.